cpp/src/gandiva/function_registry_string.cc

```diff
@@ -406,6 +406,10 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
 
       NativeFunction("split_part", {}, DataTypeVector{utf8(), utf8(), int32()}, utf8(),
                      kResultNullIfNull, "split_part",
+                     NativeFunction::kNeedsContext | NativeFunction::kCanReturnErrors),
+
+      NativeFunction("instr", {}, DataTypeVector{utf8(), utf8()},  int32(),
+                     kResultNullIfNull, "gd_fn_instr_utf8",
```

Comment:
Why does this function need context? I also dont see this function return errors

Do we need the kNeedsContext and kCanReturnErrors flags?

Reply:
This one I was waiting for review, so I just left kNeedsContext, if I remove it the projector test will fail

Reply:
This is not going to allocate memory. So, I dont understand why it needs context

Reply:
Removed kNeedsContext flag